src/pencilkit.cs

```diff
@@ -0,0 +1,243 @@
+//
+// PencilKit C# bindings
+//
+// Authors:
+//	TJ Lambert  <t-anlamb@microsoft.com>
+//
+// Copyright 2019 Microsoft Corporation All rights reserved.
+//
+
+#if MONOMAC
+using AppKit;
+using UIColor = AppKit.NSColor;
+using UIImage = AppKit.NSImage;
+
+using UIScrollViewDelegate = Foundation.NSObjectProtocol;
+using UIScrollView = Foundation.NSObject;
+using UIGestureRecognizer = Foundation.NSObject;
+using UIResponder = Foundation.NSObject;
+using UIView = Foundation.NSObject;
+using UIWindow = Foundation.NSObject;
+#else
+using UIKit;
+#endif
+
+using System;
+using ObjCRuntime;
+using Foundation;
+using CoreGraphics;
+
+namespace PencilKit {
+
+	[iOS (13, 0), NoMac]
+	[Native]
+	enum PKEraserType : long {
+		Vector,
+		Bitmap,
+	}
+
+	[iOS (13, 0), NoMac]
+	enum PKInkType {
+		[Field ("PKInkTypePen")]
+		Pen,
+
+		[Field ("PKInkTypePencil")]
+		Pencil,
+
+		[Field ("PKInkTypeMarker")]
+		Marker,
+	}
+
+	[iOS (13, 0), NoMac]
+	[BaseType (typeof (NSObject))]
+	[Model (AutoGeneratedName = true)] [Protocol]
+	interface PKCanvasViewDelegate : UIScrollViewDelegate {
+
+		[Export ("canvasViewDrawingDidChange:")]
+		void DrawingDidChange (PKCanvasView canvasView);
+
+		[Export ("canvasViewDidFinishRendering:")]
+		void DidFinishRendering (PKCanvasView canvasView);
+
+		[Export ("canvasViewDidBeginUsingTool:")]
+		void DidBeginUsingTool (PKCanvasView canvasView);
+
+		[Export ("canvasViewDidEndUsingTool:")]
+		void EndUsingTool (PKCanvasView canvasView);
+	}
+
+	interface IPKCanvasViewDelegate {}
+
+	[iOS (13, 0), NoMac]
+	[BaseType (typeof (UIScrollView))]
+	interface PKCanvasView : PKToolPickerObserver {
+
+		// This exists in the base class
+		// [Export ("delegate", ArgumentSemantic.Weak), NullAllowed]
+		// NSObject WeakDelegate { get; set; }
+
+		[Wrap ("WeakDelegate"), NullAllowed, New]
+		IPKCanvasViewDelegate Delegate { get; set; }
+
+		[Export ("drawing", ArgumentSemantic.Copy)]
+		PKDrawing Drawing { get; set; }
+
+		[Export ("tool", ArgumentSemantic.Copy)]
+		PKTool Tool { get; set; }
+
+		[Export ("rulerActive")]
+		bool RulerActive { [Bind ("isRulerActive")] get; set; }
+
+		[Export ("drawingGestureRecognizer")]
+		UIGestureRecognizer DrawingGestureRecognizer { get; }
+
+		[Export ("allowsFingerDrawing")]
+		bool AllowsFingerDrawing { get; set; }
+	}
+
+	[iOS (13, 0), Mac (10, 15, onlyOn64: true)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface PKDrawing : NSCopying, NSSecureCoding {
+
+		[Field ("PKAppleDrawingTypeIdentifier")]
+		NSString AppleDrawingTypeIdentifier { get; }
+
+		[DesignatedInitializer]
+		[Export("init")]
```

Comment:
Nitpick:

```suggestion
		[Export ("init")]
```

Reply:
This can be simplified with `[DesignatedDefaultCtor]` on the type
i.e. remove `[DisableDefaultCtor]` and `[DesignatedInitializer][Export("init")]IntPtr Constructor();`

Reply:
So just to make sure, you are suggesting that I remove the method completely and where DisableDefaultCtor is right now, I change that to DesignatedDefaultCtor?

Reply:
Also I have those DisableDefaultCtor 's there since some of them inherit from PKTool which has a disable default constructor, so is that still okay?

Reply:
@spouliot Sorry, I'm not sure if I was supposed to tag you.